lib/dependency-manager-adapters/npm.js

```diff
@@ -53,16 +77,22 @@ module.exports = CoreObject.extend({
     });
   },
   cleanup: function() {
-    var adapter = this;
-    return adapter._restoreOriginalDependencies().then(function() {
+    return this._restoreOriginalDependencies().then(() => {
       debug('Remove backup package.json and node_modules');
-      return RSVP.all([rimraf(path.join(adapter.cwd, adapter.packageJSONBackupFileName)),
-        rimraf(path.join(adapter.cwd, adapter.nodeModulesBackupLocation))]);
+
+      let cleanupTasks = [rimraf(path.join(this.cwd, this.packageJSONBackupFileName)),
+        rimraf(path.join(this.cwd, this.nodeModulesBackupLocation))];
+
+      if (this.yarnLockPresent) {
+        cleanupTasks.push(rimraf(path.join(this.cwd, this.yarnLockBackupFileName)));
+      }
+
+      return RSVP.all(cleanupTasks);
     }).catch(function(e) {
       console.log('Error cleaning up npm scenario:', e);
     })
-    .then(function() {
-      return adapter._install();
+    .then(() => {
```

Comment:
I am not familiar with that matrix but i have got an impression that `0.12` support was already dropped:
https://github.com/ember-cli/ember-try/commit/0eb4c63dc10001a042f1bf12485a4c8e4691894e#diff-b9cfc7f2cdf78a7f4b91a753d10865a2L21

so that is a little bit confusing.

I am still ok with using older syntax and would fix it.